### DISCUSSION TITLE
What is the easiest way to use another notemap.json?

### DISCUSSION THREAD

**Discussion by stardusthamster** [cardonabits/haxo-hw#54] [Q&A] [answered]
I created a new notemap.json that is tuned in C. So when I play a C3 then the sound that is played is a C3.
(I wanted to change this, as I am a flute player and I always get confused with the transposing instruments). 

What is the easiest way to use this newly created notemap_c_instrument.json instead of the included one?

Would it be possible to choose the notemap on startup? Or even by changing the config on the fly?

Any help appreciated!
Thank you very much in advance!

**Comment by jcard0na** [comment, marked as answer]
Hi @stardusthamster,

At the moment there is no way to change the notemap on startup or on the fly.  If you would like that, go ahead and [open a new issue](https://github.com/cardonabits/haxo-rs/issues/new).  To replace the existing `notemap` with your own, you can follow these steps:

1. Log into the haxophone.  There are three ways you can do that:
  a. Attach a keyboard and display and log in directly
  b. Via the [serial port on the haxophone](https://www.crowdsupply.com/cardona-bits/haxophone/updates/manufacturing-plans-and-more#linux-console-on-the-haxophone)
  c. Attach a network adapter, and log in via the network
2. Once you have a console, disable read-only mode:
```
sudo raspi-config
Performance Options -> Overlay Filesystem -> Enable Overlay? No
```
3. Copy over the new file `notemap_c_instrument.json` to `/usr/share/haxo` folder.  Again, you will have several ways to do that:
  a. Attach a usb pendrive
  b. `scp` over the network
  c. copy-paste the file contents
4. Modify [`/etc/systemd/system/haxo.service`](https://github.com/cardonabits/haxo-rs/blob/main/scripts/systemd/haxo.service#L16) to point to your new notemap
```diff
-          --notemap-file /usr/share/haxo/notemap.json \
+          --notemap-file /usr/share/haxo/notemap_c_instrument.json \
```
4. Test your change by rebooting the haxophone and confirming the new fingerings are used.
```
sudo reboot
```
5. If all works, re-enable the read-only overlay filesystem
```
sudo raspi-config
Performance Options -> Overlay Filesystem -> Enable Overlay? Yes
```
6. Optionally, if you want to contribute that notemap for others to use, make a pull request with your `notemap` file.
